modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java

```diff
@@ -96,6 +107,171 @@ public void testSchemaRegistry() {
+                .withListener(() -> "redpanda:19092")
+                .withNetwork(network);
+            GenericContainer<?> kcat = new GenericContainer<>("confluentinc/cp-kcat:7.4.1")
+                .withCreateContainerCmdModifier(cmd -> {
+                    cmd.withEntrypoint("sh");
+                })
+                .withCopyToContainer(Transferable.of("Message produced by kcat"), "/data/msgs.txt")
+                .withNetwork(network)
+                .withCommand("-c", "tail -f /dev/null")
+        ) {
+            redpanda.start();
+            kcat.start();
+            kcat.execInContainer("kcat", "-b", "redpanda:19092", "-t", "msgs", "-P", "-l", "/data/msgs.txt");
+            String stdout = kcat
+                .execInContainer("kcat", "-b", "redpanda:19092", "-C", "-t", "msgs", "-c", "1")
+                .getStdout();
+            assertThat(stdout).contains("Message produced by kcat");
+        }
+    }
+
+    @SneakyThrows
+    @Test
+    public void enableSaslWithSuccessfulTopicCreation() {
+        try (
+            RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
+                .enableAuthorization()
+                .enableSasl()
+                .withSuperuser("superuser-1")
+        ) {
+            redpanda.start();
+
+            createSuperUser(redpanda);
+
+            AdminClient adminClient = getAdminClient(redpanda);
+            String topicName = "messages-" + UUID.randomUUID();
+            Collection<NewTopic> topics = Collections.singletonList(new NewTopic(topicName, 1, (short) 1));
+            adminClient.createTopics(topics).all().get(30, TimeUnit.SECONDS);
+
+            assertThat(adminClient.listTopics().names().get()).contains(topicName);
+        }
+    }
+
+    @Test
+    public void enableSaslWithUnsuccessfulTopicCreation() {
+        try (
+            RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
+                .enableAuthorization()
+                .enableSasl()
+        ) {
+            redpanda.start();
+
+            createSuperUser(redpanda);
+
+            AdminClient adminClient = getAdminClient(redpanda);
+            String topicName = "messages-" + UUID.randomUUID();
+            Collection<NewTopic> topics = Collections.singletonList(new NewTopic(topicName, 1, (short) 1));
+
+            Awaitility
+                .await()
+                .untilAsserted(() -> {
+                    assertThatThrownBy(() -> adminClient.createTopics(topics).all().get(30, TimeUnit.SECONDS))
+                        .hasCauseInstanceOf(TopicAuthorizationException.class);
+                });
+        }
+    }
+
+    @Test
+    public void enableSaslAndWithAuthenticationError() {
+        try (
+            RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
```

Comment:
Redpanda `v23.2.2` has been recently released. Consider upgrading Redpanda version to the latest release